test/unittest/configuration/DatabrokerConfigurationTest.cpp

```diff
@@ -0,0 +1,602 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+
+#include <gtest_aux.hpp>
+#include <gtest/gtest.h>
+
+#include <databroker/configuration/DatabrokerConfiguration.hpp>
+#include <databroker/exceptions/ConfigurationException.hpp>
+#include <databroker/types/configuration_tags.hpp>
+#include <databroker/types/RawConfiguration.hpp>
+#include <databroker/types/topic/WildcardTopic.hpp>
+
+using namespace eprosima::databroker;
+
+/***************
+ * CONSTRUCTOR *
+ ***************/
+
+/*
+ * Add a topic to a list in a yaml
+ * If name or type is not given, that tag will not be added
+ */
+void add_topic_to_list_to_yaml(
+    RawConfiguration& yaml,
+    const char* list_tag,
+    std::string topic_name = "",
+    std::string topic_type = "")
+{
+    RawConfiguration topic;
+
+    if (topic_name != "")
+    {
+        topic[TOPIC_NAME_TAG] = topic_name;
+    }
+
+    if (topic_type != "")
+    {
+        topic[TOPIC_TYPE_NAME_TAG] = topic_type;
+    }
+
+    yaml[list_tag].push_back(topic);
```

Comment:
The topic will be added even if there are no name or type given.

Reply:
This could actually happen, addeded to the tests